R/compare_diagnoses.R

```diff
@@ -0,0 +1,345 @@
+
+
+
+
+#' @export
+print.compared.diagnoses <- function(object, ...) {
+  print(summary(object))
+  invisible(object)
+}
+
+
+#' @export
+summary.compared.diagnoses <- function(x, ...) {
+  structure(x, class = c("summary.compared_diagnoses", "data.frame"))
+  
+}
+
+
+
+#' @export
+print.summary.compared_diagnoses <- function(x, ...){
+  
+  bootstrap_rep1 <- x$diagnosis1$bootstrap_sims
+  bootstrap_rep2 <- x$diagnosis2$bootstrap_sims
+  n_sims1 <- nrow(x$diagnosis1$simulations_df)
+  n_sims2 <- nrow(x$diagnosis2$simulations_df)
+  x  <- x[["compared.diagnoses_df"]]
+  sx <- subset(x,  x[,"in_interval"] == 0)
+  
+  atext <- paste0("(alpha level = ", attr(x, "alpha"), ")")
+  
+  if(nrow(sx) == 0) {
+    print(paste("No divergences found ", atext)) 
+    return(x)}
+  
+  
+  if(n_sims1 == n_sims2 )
+    cat(paste0("\n Comparison of research designs diagnoses based on ", n_sims1, " simulations", atext))
+  else 
+    cat(paste0("\n Comparison of research designs diagnoses based on ", n_sims1, " simulations from `base_design` and ", n_sims2, " from `comparison_design`", atext))
+  if(bootstrap_rep1 == bootstrap_rep2)
+    cat(paste0("\nDiagnosand estimates with bootstrapped standard errors in parentheses (", bootstrap_rep1,")."  ))
+  else 
+    cat(paste0("\nDiagnosand estimates with bootstrapped standard errors in parentheses (base_design = ", bootstrap_rep1,", omparison_design = ", bootstrap_rep2, ")."  ))
+ 
+  
+  
+  se_1 <- paste0("(",format_num(sx$se_1, 2), ")")
+  
+  se_2 <- ifelse(bootstrap_rep2 !=0,
+                 paste0("(",format_num(sx$se_2, 2), ")"),
+                 rep("-", length(sx$se_1)))
+  
+ # if(bootstrap_rep2 !=0) se_2 <- paste0("(",format_num(sx$se_2, 2), ")")
+  # else se_2 <- rep(length(sx$se_1), "-")
+  
+
+  sx   <- sx[ ,!colnames(sx) %in% c("in_interval", "se_1", "se_2", "n_sims1", "n_sims2", "conf.low_1", "conf.upper_1")]
+
+  
+  # Prep vars to shape data frame
+  mand  <- startsWith(colnames(sx), "estimand")
+  term  <- startsWith(colnames(sx), "term")
+  mator <- startsWith(colnames(sx), "estimator")
+  n_labels <- sum(mand, mator, term)
+  
+  r <- nrow(sx)
+  k <- ncol(sx)
+  
+  label_cols <- matrix(replicate(n_labels + 1 , rep("", r)), 
+                       ncol =  n_labels + 1 )
+  
+  se_rows <- data.frame(label_cols,
+                        se_1,
+                        se_2,
+                        stringsAsFactors = FALSE)  
+  
+  # create a data frame double the size of diagnosands_df
+  # odd rows correspond to diagnosands' means
+  # even rows to se(diagnosands)
+  designs <- startsWith(colnames(sx), "design")
+  out <- data.frame(rbind(sx, sx), stringsAsFactors = FALSE)
+  diagnosands_cols <- startsWith(colnames(out), "mean")
+  colnames(out)[diagnosands_cols] <- c("base", "comparison")
+  m <- nrow(out)
+  odd  <- (1:m) %% 2 != 0
+  even <- (1:m) %% 2 == 0
+  diagnosands_m <- sapply(sx[,diagnosands_cols], format_num, digits=2)
+  
+  
+  out[odd, diagnosands_cols ]  <- diagnosands_m
```

Comment:
please follow reshape_diagnosis behavior here, this is a whole new set of behaviors and would be good to use the same to minimize code maintenance going forward